Mat2: implement the get(i: int, j: int) method

oybek-turaev-cyber opened the pull request:
- [x] Create a test/s for a method `get()` that returns a matrix element at a specified `i` and `j` indices
- [x] Pass the test/s for the `get()` method
- [x] Finish the default constructor test with a ToDo

oybek-turaev-cyber committed:
Mat2: Impelement get() method

oybek-turaev-cyber committed:
Mat2: Create tests for get() method

Pass Tests:

*'givenNoArgs_whenMat2Created_thenMatrixIsIdentity()'
*'givenIdentityMatrix_whenGetElement_thenReturnValueMatches()'

oybek-turaev-cyber committed:
Merge branch 'iss-14' into dev with Mat2: implement the get(i: int, j: int) method

toksaitov commented on lib/src/main/java/vmath/Mat2.java:
```diff
@@ -9,4 +9,15 @@ public Mat2() {
     }
+
+    public float get(int i, int j) {
+        
+        if (i < 0 || i >= 2 || j < 0 || j >= 2) {
+            throw new IndexOutOfBoundsException("Invalid indices");
+        }
+
+        int index = i * 2 + j;
+
+        return m[index];
```
Remove the first empty line. Organize the code in a similar manner how @dulatov-e did it [here](https://github.com/toksaitov/vmath/pull/46/files)

toksaitov commented on lib/src/test/java/vmath/Mat2Test.java:
```diff
@@ -9,6 +9,21 @@ public class Mat2Test {
+        assertEquals(1.0f, matrix.get(1, 1));
+    }
+
+    @Test
+    void givenIdentityMatrix_whenGetElement_thenReturnValueMatches() {
+        Mat2 matrix = new Mat2();
+
+        assertEquals(1.0f, matrix.get(0, 0));
+        assertEquals(0.0f, matrix.get(0, 1));
+        assertEquals(0.0f, matrix.get(1, 0));
+        assertEquals(1.0f, matrix.get(1, 1));
     }
```
Create a test to check for invalid indices. Initialize an empty test method with a TODO comment to test the `get` method. We will wait to implement this test until more complex constructors are available. See @dulatov-e's code for reference [here](https://github.com/toksaitov/vmath/pull/46/files).

oybek-turaev-cyber committed:
Mat2: Add more tests for get() method

Pass Tests:

*'givenIndices_whenGetCalled_thenReturnValueMatches()'
*'givenInvalidIndices_whenGetCalled_thenThrowIllegalArgumentException()'

oybek-turaev-cyber committed:
Merge branch 'iss-14' into dev with Mat2: implement the get(i: int, j: int) method with more tests

oybek-turaev-cyber committed:
Mat2: Remove empty line in get() method

oybek-turaev-cyber committed:
Mat2: Remove unnecessary test case for get() method

oybek-turaev-cyber committed:
Merge branch 'iss-14' into dev with Mat2: implement the get(i: int, j: int) method with fixes

rider-v commented on lib/src/test/java/vmath/Mat2Test.java:
```diff
@@ -9,6 +9,26 @@ public class Mat2Test {
-}
\ No newline at end of file
+
+    @Test
+    void givenInvalidIndices_whenGetCalled_thenThrowIllegalArgumentException() {
+        Mat2 matrix = new Mat2();
+
+        assertThrows(IndexOutOfBoundsException.class, () -> matrix.get(6, 1));
```
Please adjust the name of this test: it checks for whether get() throws IndexOutOfBoundsException, not IllegalArgumentException.

oybek-turaev-cyber replied:
I renamed this as requested

oybek-turaev-cyber committed:
Mat2: Rename the test case header for get() method

oybek-turaev-cyber committed:
Merge branch 'iss-14' into dev with Mat2: implement the get(i: int, j: int) method with fixes of renaming test case header

toksaitov commented on lib/src/main/java/vmath/Mat2.java:
```diff
@@ -9,4 +9,14 @@ public Mat2() {
+            throw new IndexOutOfBoundsException("Invalid indices");
+        }
+
+        int index = i * 2 + j;
+
+        return m[index];
```
Combine into one return statement like in other Matrix PRs.

oybek-turaev-cyber replied:
I corrected this as requested

toksaitov commented on lib/src/test/java/vmath/Mat2Test.java:
```diff
@@ -9,6 +9,26 @@ public class Mat2Test {
+        assertThrows(IndexOutOfBoundsException.class, () -> matrix.get(6, 1));
+        assertThrows(IndexOutOfBoundsException.class, () -> matrix.get(2, 1));
+        assertThrows(IndexOutOfBoundsException.class, () -> matrix.get(-1, 1));
+        assertThrows(IndexOutOfBoundsException.class, () -> matrix.get(-1, 13));
```
Let's check the invalid values on boundaries like in other PRs.

oybek-turaev-cyber replied:
I corrected this as requested

oybek-turaev-cyber committed:
Mat2: Improve code structure in get() method

oybek-turaev-cyber committed:
Mat2: Check Invalid Values with better cases for get() method

oybek-turaev-cyber committed:
Merge branch 'iss-14' into dev with Mat2: implement the get(i: int, j: int) method with improving code structure

inchina-k commented on lib/src/main/java/vmath/Mat2.java:
```diff
@@ -9,4 +9,12 @@ public Mat2() {
+            throw new IndexOutOfBoundsException("Invalid indices");
+        }
+
+        return m[i * 2 + j];
```
The code style is incorrect, it should be m[i*2 + j]
